tools/backups2datalad/aioutil.py

```diff
@@ -238,29 +209,117 @@ async def stream_null_command(
     if cwd is not None:
         desc += f" [cwd={cwd}]"
     log.debug("Opening pipe to %s", desc)
-    async with await anyio.open_process(argstrs, cwd=cwd, stderr=None) as p:
-        buff = ""
+    async with kill_on_error(
+        await anyio.open_process(argstrs, cwd=cwd, stderr=None)
+    ) as p:
         assert p.stdout is not None
-        async for text in TextReceiveStream(p.stdout):
+        async with TextReceiveStream(p.stdout) as stream:
             try:
-                buff += text
-                while True:
-                    try:
-                        i = buff.index("\0")
-                    except ValueError:
-                        break
-                    else:
-                        yield buff[:i]
-                        buff = buff[i + 1 :]
+                splitter = TerminatedSplitter("\0", retain=False)
+                async for chunk in splitter.aitersplit(stream):
+                    yield chunk
             except BaseException:
                 log.exception("Exception raised while handling output from %s", desc)
                 raise
-        if buff:
-            yield buff
     log.log(
         logging.DEBUG if p.returncode == 0 else logging.WARNING,
         "Command %s exited with return code %d",
         desc,
         p.returncode,
     )
     ### TODO: Raise an exception if p.returncode is nonzero?
+
+
+async def stream_lines_command(
+    *args: str | Path, cwd: Optional[Path] = None
+) -> AsyncGenerator[str, None]:
+    argstrs = [str(a) for a in args]
+    desc = f"`{shlex.join(argstrs)}`"
+    if cwd is not None:
+        desc += f" [cwd={cwd}]"
+    log.debug("Opening pipe to %s", desc)
+    async with kill_on_error(
+        await anyio.open_process(argstrs, cwd=cwd, stderr=None)
+    ) as p:
+        assert p.stdout is not None
+        async for line in LineReceiveStream(TextReceiveStream(p.stdout)):
+            yield line
+    log.log(
+        logging.DEBUG if p.returncode == 0 else logging.WARNING,
+        "Command %s exited with return code %d",
+        desc,
+        p.returncode,
+    )
+    ### TODO: Raise an exception if p.returncode is nonzero?
```

Comment:
I guess we have exception now, and we better have exception so we do not miss unexpected crashes... if it is expected -- should then be handled explicitly (outside or though the mode option here)

Reply:
What exactly are you suggesting?

Reply:
I see now that it is a result of copy pasting from prior function.
I guess nevermind for now, but I thought to have here
```python
if p.returncode:
    raise RuntimeError(f"Command {desc} exited with return code {p.returncode}")
```
here